x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -281,15 +281,15 @@ export const resolve = async (
       return CaseResolveResponseRt.encode({
         ...resolveData,
         case: flattenCaseSavedObject({
-          savedObject,
+          savedObject: resolvedSavedObject,
           subCaseIds,
         }),
       });
     }
 
     const theComments = await caseService.getAllCaseComments({
       unsecuredSavedObjectsClient,
-      id,
+      id: resolvedSavedObject.id,
```

Comment:
In the case of an alias match the request's `id` and the `resolvedSavedObject.id` will be different. We'll want to use the `resolvedSavedObject.id` instead since that'll be pointing to the changed id.